src/core/Akka.Remote/EndpointManager.cs

```diff
@@ -324,12 +330,39 @@ private ICancelable PruneTimerCancelleable
                 {
                     return _pruneTimeCancelable = Context.System.Scheduler.ScheduleTellRepeatedlyCancelable(PruneInterval, PruneInterval, Self, new Prune(), Self);
                 }
-                return null;
+                return _pruneTimeCancelable;
+            }
+        }
+
+        private Dictionary<IActorRef, AkkaProtocolHandle> _pendingReadHandoffs = new Dictionary<IActorRef, AkkaProtocolHandle>();
+        private Dictionary<IActorRef, List<InboundAssociation>> _stashedInbound = new Dictionary<IActorRef, List<InboundAssociation>>();
+
+
+        private void HandleStashedInbound(IActorRef endpoint, bool writerIsIdle)
+        {
+            var stashed = _stashedInbound.GetOrElse(endpoint, new List<InboundAssociation>());
+            _stashedInbound.Remove(endpoint);
+            foreach (var ia in stashed)
+                HandleInboundAssociation(ia, writerIsIdle);
+        }
+
+        private void KeepQuarantinedOr(Address remoteAddress, Action body)
```

Comment:
New utility method used by the `SupervisorStrategy` for `EndpointManager`. 

Works via the following:
1. If we have a refusable UID for the `remoteAddress` inside the `EndpointRegistry`, we will keep the quarantine but overwrite it with a new `Quarantine` policy.
2. Otherwise, we'll action the `Action` delegate passed to the method.